Lib/collections/__init__.py

```diff
@@ -410,26 +360,79 @@ def namedtuple(typename, field_names, *, verbose=False, rename=False, module=Non
+    _len = len
+
+    # Create all the named tuple methods to be added to the class namespace
+
+    def create_new():
+        s = f'def __new__(_cls, {arg_list}): return _tuple_new(_cls, ({arg_list}))'
+        namespace = dict(_tuple_new=tuple_new, __name__=f'namedtuple_{typename}')
+        # Note: exec() has the side-effect of interning the typename and field names
+        exec(s, namespace)
+        __new__ = namespace['__new__']
+        __new__.__doc__ = f'Create new instance of {typename}({arg_list})'
+        return __new__
+
+    def create_make():
+        def _make(cls, iterable):
+            result = tuple_new(cls, iterable)
+            if _len(result) != num_fields:
+                raise TypeError(f'Expected {num_fields:d} arguments, got {len(result)}')
+            return result
+        _make.__doc__ = f'Make a new {typename} object from a sequence or iterable'
+        return classmethod(_make)
+
+    def create_replace():
+        def _replace(_self, **kwds):
+            result = _self._make(map(kwds.pop, field_names, _self))
+            if kwds:
+                raise ValueError(f'Got unexpected field names: {list(kwds)!r}')
+            return result
+        _replace.__doc__ = f'Return a new {typename} object replacing specified fields with new values'
+        return _replace
 
-    # Execute the template string in a temporary namespace and support
-    # tracing utilities by setting a value for frame.f_globals['__name__']
-    namespace = dict(__name__='namedtuple_%s' % typename)
-    exec(class_definition, namespace)
-    result = namespace[typename]
-    result._source = class_definition
-    if verbose:
-        print(result._source)
+    def __repr__(self):
+        'Return a nicely formatted representation string'
+        return self.__class__.__name__ + repr_fmt % self
+
+    def _asdict(self):
+        'Return a new OrderedDict which maps field names to their values.'
+        return OrderedDict(zip(self._fields, self))
+
+    def __getnewargs__(self):
+        'Return self as a plain tuple.  Used by copy and pickle.'
+        return tuple(self)
+
+    def reuse_itemgetter(index):
+        try:
+            return _nt_itemgetters[index]
+        except KeyError:
+            getter = _nt_itemgetters[index] = _itemgetter(index)
+            return getter
+
+    # Build-up the class namespace dictionary
+    # and use type() to build the result class
+    class_namespace = dict(
+        __slots__ = (),
+        __doc__ = f'{typename}({arg_list})',
+        _fields = field_names,
+        __new__ = create_new(),
+        _make = create_make(),
+        _replace = create_replace(),
+        __repr__ = __repr__,
+        _asdict = _asdict,
+        __getnewargs__ = __getnewargs__,
+    )
+    for index, name in enumerate(field_names):
+        class_namespace[name] = property(fget = reuse_itemgetter(index),
+                                         doc = f'Alias for field number {index}')
```

Comment:
Nice hack! How about reusing docstring too?

Reply:
Yes, that is a great idea.  Adding:   doc = _sys.intern(f'Alias for field number {index}')